src/components/AlbumForm.vue

```diff
@@ -375,52 +375,44 @@ export default {
         album_artists: [],
       };
 
-      const promises = [];
-
-      for (let label of this.newAlbum.album_labels) {
+      const label_map = this.newAlbum.album_labels.map(async (label) => {
```

Comment:
```suggestion
      const mappedLabels = this.newAlbum.album_labels.map(async (label) => {
``` 
Also note the use of `camelCase`; we only use `snake_case` for values in the store that come directly from the API. This should be fixed in a lot of places.